glam/api/migrations/0022_create_backup_MV.py

```diff
@@ -0,0 +1,82 @@
+# Generated by Django 3.1.13 on 2022-03-04 01:21
+
+from django.db import migrations, models
+
+from glam.api import constants
+
+sql_backup = []
+for channel in constants.CHANNEL_NAMES.values():
+    sql_backup.extend(
+        [
+            f"CREATE MATERIALIZED VIEW view_glam_desktop_{channel}_aggregation_backup AS SELECT * FROM glam_desktop_{channel}_aggregation;",
+            f"CREATE UNIQUE INDEX ON view_glam_desktop_{channel}_aggregation_backup (id);",
+            f"CREATE INDEX ON view_glam_desktop_{channel}_aggregation_backup (version);",
+            f"CREATE INDEX ON view_glam_desktop_{channel}_aggregation_backup USING HASH (metric);",
+            f"CREATE INDEX ON view_glam_desktop_{channel}_aggregation_backup USING HASH (os);",
+        ]
+    )
```

Comment:
This code runs at import time. You probably want to put this in a function so that it doesn't run every time this file is imported.

Also, it's generally better to write migrations such that they're "set in stone" at the time you wrote/ran them. This one uses `glam.api.constants`. If that changes in the future, then this migration changes.

Reply:
Sorry,  I did not understand this. Can you please elaborate?
The backup MVs will not exist when running the import command. They are dropped in the next migration command. (glam/api/migrations/0023_drop_alter_MV.py)

Reply:
I should have looked at other migrations in this app before saying anything. While I tend to write migrations in a way that they're "set in stone" because I feel that's safer and it's bit me in the past, it looks like glam does things differently and that's cool.

Also, if someone squashes the migrations at some point, then all of this is moot.